media/js/base/translation-bar.es6.js

```diff
@@ -0,0 +1,110 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+(function () {
+    const barElem = document.getElementById('translation-bar');
+    if (!barElem) {
+        return false;
+    }
+
+    // Normalize language in the form of ab or ab-CD.
+    function normalize(lang) {
+        return lang.replace(/^(\w+)(?:-(\w+))?$/, function (m, p1, p2) {
+            return p1.toLowerCase() + (p2 ? '-' + p2.toUpperCase() : '');
+        });
+    }
+
+    // Returns boolean if lang matches target lang.
+    function isLangMatch(targetLang, lang) {
+        return (
+            targetLang === lang ||
+            targetLang === lang.split('-')[0] ||
+            targetLang.split('-')[0] === lang.split('-')[0]
+        );
+    }
+
+    // Return best matched available language, or `false` if none found.
+    function getBestMatch(acceptLangs, availableLangs) {
```

Comment:
Suggestion: it would be good to maintain some of the original [tests](https://github.com/mozilla/mozilla-infobar/blob/master/tests/unit/spec/infobar.js) we had for this feature.

A nice way to make the code here more easily testable might be to separate the transbar logic from the code that initialises it to run in the page. Making the transbar an ES module which another file can import and run is a good way to do this perhaps, as it also means a test file can import and test the the module in the same way.